scm-spain/boros-CMP
# Fix the CMP_VERSION assigned on build process

## Description


Actually, the cmpVersion stored in the consent strings are "0" due to a bad implementation of the defined property that should be replaced in the build process.

The README description tells the intention of having the project's major number as the used cmpVersion number. This PR fixes that, creating a new "internal" property that is automatically filled via babel-plugin, being:
* The major version of the package.json, converted to int

## Solves ticket/s


* Issue: cmpVersion is not set correctly #67 
* https://jira.scmspain.com/browse/PSP-2518

## Expected behavior


* A new consent has the _"cmpVersion": 1_

## Review steps


* Run
  ```
  npm run start:cmp:dev
  ```

* Open the localhost link shown in the console

* Now, in the browser's console, run:
  ```
  window.__cmp('setVendorConsents', {vendorConsents: [], purposeConsents: []}, function(){})
  ```

* Copy the 'euconsent' cookie value and decode it (p.ex. at <https://www.decodeconsentstring.com>)

* The cmpVersion should be 1

Sample decoded consent after the fix:
* <https://www.decodeconsentstring.com/v1.0/?consent-string=BOl9VxQOl9VxQCBABBESCiAAAAAp6AAA>

## Further considerations


* The *babel-preset-sui* has been downgraded to *3.0.0* due to an issue with its dependencies that was causing errors on tests which were using classes with an "extends". Will be upgraded when solved, or be replaced to the *preset-env*

## Memetized description

![mandatory](https://media.giphy.com/media/l41YtZOb9EUABnuqA/giphy.gif)
